sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Canonicalize.scala

```diff
@@ -41,6 +41,7 @@ object Canonicalize {
   private[expressions] def ignoreNamesTypes(e: Expression): Expression = e match {
     case a: AttributeReference =>
       AttributeReference("none", a.dataType.asNullable)(exprId = a.exprId)
+    case GetStructField(child, ordinal, Some(_)) => GetStructField(child, ordinal, None)
```

Comment:
It looks not precisely matched the comments of `ignoreNamesTypes`. It's better to change it accordingly.

Reply:
`  /** Remove names and nullability from types. */`

I can change it to `and / or`.

Reply:
The comment of `Canonicalize` says:

```
The following rules are applied:
 *  - Names and nullability hints for [[org.apache.spark.sql.types.DataType]]s are stripped.
...
```

It's also needed to be update.

Reply:
> ` /** Remove names and nullability from types. */`

Actually after this change it is not only for types.